modules/tinymce/src/plugins/imagetools/test/ts/browser/ImageToolsDisabledButtonsTest.ts

```diff
@@ -0,0 +1,74 @@
+import { Mouse, UiFinder } from '@ephox/agar';
+import { before, describe, it } from '@ephox/bedrock-client';
+import { McEditor, TinyDom } from '@ephox/mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import ImagePlugin from 'tinymce/plugins/image/Plugin';
+import ImageToolsPlugin from 'tinymce/plugins/imagetools/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.plugins.imagetools.ImageToolsDisabledButtonsTest', () => {
+
+  const settings = {
+    plugins: 'image imagetools',
+    toolbar: 'editimage fliph flipv rotateleft rotateright',
+    base_url: '/project/tinymce/js/tinymce',
+    height: 900
+  };
+
+  const editImageButtonSelector = '[role="toolbar"] button[title="Edit image"]';
+  const flipHorizontallyImageButtonSelector = '[role="toolbar"] button[title="Flip horizontally"]';
+  const flipVerticallyImageButtonSelector = '[role="toolbar"] button[title="Flip vertically"]';
+  const rotateCounterClockwiseImageButtonSelector = '[role="toolbar"] button[title="Rotate counterclockwise"]';
+  const rotateClockwiseImageButtonSelector = '[role="toolbar"] button[title="Rotate clockwise"]';
+
+  before(() => {
+    Theme();
+    ImageToolsPlugin();
+    ImagePlugin();
+  });
+
+  const assertButtonsStatus = (editor: Editor, disabled: boolean) => {
+    const container = TinyDom.container(editor);
+    UiFinder.exists(container, `${editImageButtonSelector}[aria-disabled="${disabled}"]`);
+    UiFinder.exists(container, `${flipHorizontallyImageButtonSelector}[aria-disabled="${disabled}"]`);
+    UiFinder.exists(container, `${flipVerticallyImageButtonSelector}[aria-disabled="${disabled}"]`);
+    UiFinder.exists(container, `${rotateCounterClockwiseImageButtonSelector}[aria-disabled="${disabled}"]`);
+    UiFinder.exists(container, `${rotateClockwiseImageButtonSelector}[aria-disabled="${disabled}"]`);
+  };
+
+  const insertRemoteImage = (editor: Editor) => {
+    const remoteImageSrc = 'https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png';
+    editor.setContent(`<p><img src="${remoteImageSrc}" width="50" height="50" /></p>`);
+  };
+
+  it('TINY-7772: Edit, flip and rotate should be disabled for remote images without imagetools_proxy set', async () => {
+    const editor = await McEditor.pFromSettings<Editor>(settings);
+    insertRemoteImage(editor);
+    Mouse.clickOn(TinyDom.body(editor), 'img');
+
+    const disabled = true;
+    assertButtonsStatus(editor, disabled);
```

Comment:
You don't need this variable any more as you've moved the logic out into a separate function
```suggestion
    assertButtonsStatus(editor, true);
```

Reply:
I use the variable to be easier to understand. When you pass `true` it doesn't mean much...

Reply:
I understand, but this isn't something we do anywhere else and I'd like us to be consistent. If you're worried about readability then we'd normally pass a "spec" (object) instead so you can do something like `assertButtonStatus(editor, { disabled: true })`

Reply:
Yeah, I was going to ask that `{disabled: boolean}`. Great, I'm changing 👍